Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/XDRPolicyTransformHelperTest.java

```diff
@@ -0,0 +1,227 @@
+import oasis.names.tc.xacml._2_0.context.schema.os.AttributeType;
+import oasis.names.tc.xacml._2_0.context.schema.os.RequestType;
+import oasis.names.tc.xacml._2_0.context.schema.os.SubjectType;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+//@RunWith(JMock.class)
+
+public class XDRPolicyTransformHelperTest {
+
+	private static final String MESSAGE_ID = "12345566";
+	private XDRPolicyTransformHelper xdrTransformHelper;
+	private static final String TEST_HC_VAL = "urn:oid:1.1.1.1.1.1.1";
+
+	public XDRPolicyTransformHelperTest() {
+	}
+
+	@Before
+	public void setUp() {
+		xdrTransformHelper = new XDRPolicyTransformHelper();
+	}
+
+	@After
+	public void tearDown() {
+		xdrTransformHelper = null;
+	}
+
+	@Test
+	public void testTransformXDRToCheckPolicy() throws Exception {
+
+		XDREventType eventA = new XDREventType();
+
+		HomeCommunityType hct = new HomeCommunityType();
+		hct.setHomeCommunityId(TEST_HC_VAL);
+		hct.setName("bcd");
+		hct.setDescription("jkl");
+		AssertionType assertionTypeTest = new AssertionType();
+		assertionTypeTest.setMessageId(MESSAGE_ID);
+		SlotType1 slotType1 = new SlotType1();
+		slotType1.setName("$XDSDocumentEntryPatientId");
+
+		ValueListType valueListType = new ValueListType();
+		valueListType.getValue().add("1111.4444^^^&amp;26.489.22&amp;ISO");
+		slotType1.setValueList(valueListType);
+
+		PersonNameType pnt = new PersonNameType();
+		pnt.setFullName("abc");
+		assertionTypeTest.setPersonName(pnt);
+		SlotListType rsl = new SlotListType();
+
+		rsl.getSlot().add(slotType1);
+
+		SubmitObjectsRequest sor = new SubmitObjectsRequest();
+		sor.setId("100");
+		sor.setRequestSlotList(rsl);
+
+		ProvideAndRegisterDocumentSetRequestType prdsrt = new ProvideAndRegisterDocumentSetRequestType();
+		Document doc = new Document();
+		doc.setId("223");
+		prdsrt.getDocument().add(doc);
+		prdsrt.setSubmitObjectsRequest(sor);
+
+		RegistryObjectListType rolt = new RegistryObjectListType();
+
+		sor.setRegistryObjectList(rolt);
+
+		XDRMessageType xmt = new XDRMessageType();
+		eventA.setMessage(xmt);
+		xmt.setAssertion(assertionTypeTest);
+		xmt.setProvideAndRegisterDocumentSetRequest(prdsrt);
+
+		RequestType request = new RequestType();
+		SubjectHelper subjHelp = new SubjectHelper();
+		SubjectType subject = subjHelp.subjectFactory(hct, assertionTypeTest);
+		request.getSubject().add(subject);
+
+		assertionTypeTest.setHomeCommunity(hct);
+		assertionTypeTest.setSSN("170112756");
+		eventA.setSendingHomeCommunity(hct);
+		String direction = "inbound";
+		eventA.setDirection(direction);
+		eventA.getMessage().setAssertion(assertionTypeTest);
+
+		CheckPolicyRequestType result = xdrTransformHelper.transformXDRToCheckPolicy(eventA);
+
+		verifycheckPolicy_Request(result, direction);
+
+	}
+
+	@Test
+	public void testTransformXDRResponseToCheckPolicy() throws Exception {
+
+		XDRResponseEventType eventA = new XDRResponseEventType();
+
+		HomeCommunityType hct = new HomeCommunityType();
```

Comment:
Duplicate block for set up test data.  consider to consolidate into 1 common set up.